src/core/useAnimations.tsx

```diff
@@ -16,11 +16,13 @@ export function useAnimations<T extends AnimationClip>(
 ): Api<T> {
   const ref = React.useRef<Object3D>()
   const [actualRef] = React.useState(() => (root ? (root instanceof Object3D ? { current: root } : root) : ref))
+  React.useEffect(() => void (actualRef.current = root && (root instanceof Object3D ? root : root.current)), [root])
```

Comment:
This needs to be a `useLayoutEffect` for L22.

Reply:
Yeah, do you see a reason for using a layout effect on L22? If not I'll just move L19 and L22 into the same effect.

Reply:
Layout effects happen before the first frame is rendered whereas regular effects are async and will be at least a frame late. You can combine them in any case if you want.